src/lib/users/clients/inst_user_first.rb

```diff
@@ -0,0 +1,699 @@
+          else
+            retval = :notnext
+            next
+          end
+        end
+        break if retval == :cancel || retval == :accept || retval == :back
+      end
+
+      if retval == :accept
+        @encryption_method = UI.QueryWidget(Id(:encryption_method), :CurrentButton)
+        UsersSimple.SetEncryptionMethod(@encryption_method)
+      end
+      Wizard.CloseDialog
+      retval == :accept
+    end
+
+    # build the term with current user configuration status
+    def get_status_term
+      # summary label
+      auth_line = _("The authentication method is local /etc/passwd.")
+
+      # summary label
+      details_line = _("The password encryption method is %s.") %
+        @encoding2label.fetch(@encryption_method, @encryption_method)
+
+      imported_term = Empty()
+
+      if !@to_import.empty?
+        # summary label, %s is a single user name or multiple usernames (comma separated)
+        imported = n_("User %s will be imported.", "Users %s will be imported.", @to_import.size) % @to_import.join(",")
+
+        if @text_mode
+          auth_line += "<br>" + imported
+        else
+          imported_term = Left(Label(imported))
+        end
+      end
+
+      status = @text_mode ?
+        RichText(auth_line + "<br>" + details_line) :
+        VBox(Left(Label(auth_line)), imported_term, Left(Label(details_line)))
+
+      button = HBox(
+        # pushbutton label
+        Right(PushButton(Id(:change), _("&Change...")))
+      )
+
+      @text_mode ?
+        VBox(status, button) :
+        # frame label
+        Frame(
+          _("Summary"),
+          HBox(HSpacing(0.2), VBox(status, button, VSpacing(0.2)))
+        )
+    end
+
+    def valid_username?(username)
+      error = UsersSimple.CheckUsernameLength(username)
+      if !error.empty?
+        Report.Error(error)
+        return false
+      end
+
+      error = UsersSimple.CheckUsernameContents(username, "")
+      if !error.empty?
+        Report.Error(error)
+        return false
+      end
+
+      error = UsersSimple.CheckUsernameConflicts(username)
+      if !error.empty?
+        Report.Error(error)
+        return false
+      end
+
+      return true
+    end
+
+    def valid_password?(username, pw1, pw2)
+      if pw1 != pw2
+        # The two group password information do not match
+        # error popup
+        Report.Error(_("The passwords do not match.\nTry again."))
+        return false
+      end
+
+      error = UsersSimple.CheckPassword(pw1, "local")
+      if !error.empty?
+        Report.Error(error)
+        return false
+      end
+
+      passwd = Users::LocalPassword.new(uid: username, plain: pw1, also_for_root: @use_pw_for_root)
+      if !passwd.valid?
+        message = passwd.errors.join("\n\n") + "\n\n" + _("Really use this password?")
+        if !Popup.YesNo(message)
+          return false
+        end
+      end
+
+      return true
+    end
+
+  end unless defined? InstUserFirstClient
```

Comment:
no longer needed as you have it as lib and not client, so it won't be reevaluated unless some use `load` instead of `require`